Update to nim 1.4.0

squattingmonk opened the pull request:
This fixes #25. Nim added UDP support to `asyncnet` in 1.4.0, which conflicted with `private/asyncnetudp`. I've changed it to use `asyncnet`. I don't know how to thoroughly test the changes, but the following yields the expected values:

```
$ nwn_net 145.239.204.29:5122 bnds
{
  "port": 5122,
  "serverDescription": "",
  "moduleDescription": "Arelith is a roleplaying server set in the Forgotten Realms circa 1372 DR. No custom content needed; new players are most welcome! Check out arelith.com for more information.",
  "serverVersion": "8193",
  "gameType": 3,
  "_querymsec": 144,
  "_queryhost": "145.239.204.29",
  "_queryport": 5122,
  "_localport": 39990,
  "_localhost": "0.0.0.0"
}
```

squattingmonk committed:
Inherit from more specific exception classes

This fixes the warning message on compilation with nim >= 1.4.0.

squattingmonk committed:
Remove unused imports

squattingmonk committed:
Use exitprocs.addExitProc over deprecated addQuitProc

squattingmonk committed:
Silence LockLevel warnings

squattingmonk committed:
Use asyncnet for UDP requests

Nim 1.4.0 added UDP support to asyncnet. This fixes a compilation error
under 1.4.0.

niv commented:
I didn't want to update the root dependency to 1.4.0. 1.0.8 is still widely used, not the least of which company-internal.

Thank you for your PR though. I'll take a close look as soon as I can and do it in a backwards-compatible fashion.

squattingmonk committed:
Roll back required nim version to 1.0.8

Adds version checks to allow compilation under nim 1.4.0 while still
supporting the previously required 1.0.8.

squattingmonk commented:
I rolled back to 1.0.8 and added some version checks for the 1.4.0-specific stuff. Compiles and runs against both versions, but I'm sure it needs some more rigorous testing.

niv commented:
Thank you for your work and help here. I tried it on 1.0.8 and everything seems OK-ish. The diff also is easy on the eyes, so I'm good with this change.
